server/src/server.ts

```diff
@@ -55,57 +60,91 @@ connection.onInitialize((params: InitializeParams) => {
     return result;
 });
 
+// really start all functionality, like watching the workspace, after the client has initialized.
 connection.onInitialized(() => {
+    shutdownCheck();
+    workspaceLookup = new WorkspaceLookup(connection.workspace);
+    omtLinkProvider = new OMTLinkProvider(workspaceLookup);
+    workspaceLookup.init().then(() => {
+        workspaceLookup.scanAll();
+    });
+
     if (hasConfigurationCapability) {
         // Register for all configuration changes.
         connection.client.register(DidChangeConfigurationNotification.type, undefined);
     }
-    if (hasWorkspaceFolderCapability) {
-        connection.workspace.onDidChangeWorkspaceFolders(_event => {
-            connection.console.log('Workspace folder change event received.');
-        });
+    if (hasDocumentLinkCapabilities) {
+        connection.onDocumentLinks(documentLinksHandler);
+        connection.onDocumentLinkResolve(documentLinkResolve);
     }
 });
 
-// The example settings
-interface ExampleSettings {
-    maxNumberOfProblems: number;
-}
-
-// The global settings, used when the `workspace/configuration` request is not supported by the client.
-// Please note that this is not the case when using this server with the client provided in this example
-// but could happen with other clients.
-const defaultSettings: ExampleSettings = { maxNumberOfProblems: 1000 };
-let globalSettings: ExampleSettings = defaultSettings;
-
-// Cache the settings of all open documents
-let documentSettings: Map<string, Thenable<ExampleSettings>> = new Map();
+/* shutdown protocol
```

Comment:
```suggestion
/**
 * Shutdown protocol:
```

Reply:
I did not mean for this to be documentation for the boolean, but it is better and usefull, so I'll add it to the next commit.